index.js

```diff
@@ -16,59 +14,279 @@ const serializeError = FJS({
+  globalParams.globalTimeWindow = typeof settings.timeWindow === 'string'
+    ? ms(settings.timeWindow)
+    : typeof settings.timeWindow === 'number'
+      ? settings.timeWindow
       : 1000 * 60
 
-  const store = opts.redis
-    ? new RedisStore(opts.redis, timeWindow)
-    : new LocalStore(timeWindow, opts.cache)
+  // define the name of the app component. Related to redis, it will be use as a part of the keyname define in redis.
+  const pluginComponent = {
+    app: settings.appName || 'default'
+  }
+
+  // define the global behavior for the error;
+  globalParams.skipOnError = settings.skipOnError === true
+
+  // if we use redis then ...
+  if (settings.redis) {
+    // create a new redisStore connection
+    pluginComponent.store = new RedisStore(settings.redis)
+
+    // if the whitelist must be store in redis then ...
+    if (settings.whiteListInRedis) {
+      // add to the "global" whitelist, the "key" define in the "whitelist" object.
+      pluginComponent.store.addWhiteList(`${pluginComponent.app}:wlg`, settings.whitelist || [], (err) => {
+        if (err && globalParams.skipOnError === false) return next(err)
+      })
 
-  const keyGenerator = typeof opts.keyGenerator === 'function'
-    ? opts.keyGenerator
+      // we indicate that the plugin use redis and that the whitelist are managed in redis too. This is useful for the next step when creating the "preHandler"
+      pluginComponent.isRedis = true
+      pluginComponent.whiteListInRedis = true
+    } else {
+      // as we use redis, but we don't want to store the whitelist in redis, we define the "global" whitelist in the "pluginComponent"
+      pluginComponent.whitelist = {
+        global: settings.whitelist || [],
+        endpoint: []
+      }
+      // we indicate that the plugin use redis but that the whitelist are managed locally (LRU). This is useful for the next step when creating the "preHandler"
+      pluginComponent.isRedis = true
+      pluginComponent.whiteListInRedis = false
+    }
+  } else {
+    // As we don't use redis at all, we define LRU connector, we set the "global" whitelist in a object. Same for the "endpoint" whitelist that will be populated during the creation of the "preHandler"
+    pluginComponent.store = new LocalStore(settings.cache)
+    pluginComponent.isRedis = false
+    pluginComponent.whiteListInRedis = false
+    pluginComponent.whitelist = {
+      global: settings.whitelist || [],
+      endpoint: {}
+    }
+  }
+
+  // we define the default "after" to use
+  globalParams.after = ms(globalParams.globalTimeWindow, { long: true })
+
+  globalParams.keyGenerator = typeof settings.keyGenerator === 'function'
+    ? settings.keyGenerator
     : (req) => req.raw.ip
 
-  const skipOnError = opts.skipOnError === true
-  const max = opts.max || 1000
-  const whitelist = opts.whitelist || []
-  const after = ms(timeWindow, { long: true })
+  // const makeParams = (p) => { return { ...globalParams, ...p } }
```

Comment:
Why is this ommented out?

Reply:
It was crashing during the test on a older version of node.